app/models/user.rb

```diff
@@ -5,32 +5,56 @@ class User < ApplicationRecord
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :trackable, :validatable
 
+
   def has_membership_application?
     membership_applications.size > 0
   end
 
+
   def has_company?
     has_membership_application? && (membership_applications.select { |app| app.company }).count > 0
   end
 
+
   def membership_application
     has_membership_application? ? membership_applications.last : nil
   end
 
+
   def company
     has_company? ? membership_application.company : nil
   end
 
+
   def admin?
     admin
   end
 
+
   def is_member?
     is_member
   end
 
+
   def is_member_or_admin?
     admin? || is_member?
   end
 
+
+  def is_in_company_numbered?(company_num)
+    is_member? && !(companies.detect { |c| c.company_number == company_num }).nil?
```

Comment:
I know I am late on this but .... this line is a bit difficult to read - consider something like:
```
is_member? && ! companies.where(company_number: company_num ).empty?
```